pkg/network/server.go

```diff
@@ -517,6 +517,33 @@ func (s *Server) handleInvCmd(p Peer, inv *payload.Inventory) error {
 	return nil
 }
 
+// handleMempoolCmd handles getmempool command.
+func (s *Server) handleMempoolCmd(p Peer) error {
+	txs := s.chain.GetMemPool().GetVerifiedTransactions()
+	sz := len(txs)
+	if sz > payload.MaxHashesCount {
+		sz = payload.MaxHashesCount
```

Comment:
I don't think it's the way it should work. You're limiting your answer to `MaxHashesCount` transactions and then sending an `inv` for each transaction which doesn't make much sense. Instead you should be creating as much `inv` messages as needed with `MaxHashesCount` hashes inside each to transmit all hashes from the mempool.

Reply:
No, I append hash of every transaction in the loop, sending if length exceeds `MaxHashesCount`. This if is purely for allocating precise amount of memory needed. I removed this as this case is rare and obviously hurt readability.